pkg/virt-launcher/virtwrap/network/network_test.go

```diff
@@ -31,7 +31,12 @@ import (
 
 var _ = Describe("VMNetworkConfigurator", func() {
 	Context("interface configuration", func() {
-
+		resetPodNetworkConfiguratorFactory := func(nics []podNIC) []podNIC {
+			for i, _ := range nics {
+				nics[i].podNetworkConfiguratorFactory = nil
+			}
+			return nics
+		}
```

Comment:
At the end I have change the approach of mocking the network configurator and created a unexported Phase1 function used only for unit tests with a network configurator argument, the alternative approach of extra constructor does not fit well since network configurators depends on launcherPID and is true that we can check that in the constructor and all but passing the mock at the place where it's used make more sense.